src/styled.js

```diff
@@ -1,29 +1,37 @@
 import { css } from './css';
 
-let h;
-const setPragma = pragma => (h = pragma);
+let h, forwardRef, useTheme;
+const setup = (_h, _forwardRef, _useTheme = () => {}) => {
+    h = _h;
+    forwardRef = _forwardRef;
+    useTheme = _useTheme;
+};
 
 /**
  * Styled function
  * @param {String} tag
  */
 function styled(tag) {
     const _ctx = this || {};
-
     return function() {
         const _args = arguments;
-
-        return function Styled(props) {
-            const _props = (_ctx.p = Object.assign({}, props));
+        function Styled(props, ref) {
+            const theme = useTheme();
+            const _props = (_ctx.p = Object.assign(theme ? { theme } : {}, props));
             const _previousClassName = _props.className;
 
             _ctx.o = /\s*go[0-9]+/g.test(_previousClassName);
             _props.className =
                 css.apply(_ctx, _args) + (_previousClassName ? ' ' + _previousClassName : '');
+            if (ref) {
+                _props.ref = ref;
```

Comment:
If we drop the `if` and always do `_props.ref = ref;` would that be strange? 🤔

Reply:
In terms of functionality, I don't think it'd be any different. As `ref` is treated as special props, it won't showed up in component props anyway.